packages/dd-trace/src/aiguard/sdk.js

```diff
@@ -0,0 +1,164 @@
+'use strict'
+
+const NoopAIGuard = require('./noop')
+const executeRequest = require('./client')
+const {
+  AI_GUARD_RESOURCE,
+  AI_GUARD_TARGET_TAG_KEY,
+  AI_GUARD_REASON_TAG_KEY,
+  AI_GUARD_ACTION_TAG_KEY,
+  AI_GUARD_BLOCKED_TAG_KEY,
+  AI_GUARD_META_STRUCT_KEY,
+  AI_GUARD_TOOL_NAME_TAG_KEY
+} = require('./tags')
+const log = require('../log')
+
+const ALLOW = 'ALLOW'
+
+class AIGuardAbortError extends Error {
+  constructor (reason) {
+    super(reason)
+    this.name = 'AIGuardAbortError'
+    this.reason = reason
+  }
+}
+
+class AIGuardClientError extends Error {
+  constructor (message, opts = {}) {
+    super(message)
+    this.name = 'AIGuardClientError'
+    if (opts.errors) {
+      this.errors = opts.errors
+    }
+    if (opts.cause) {
+      this.cause = opts.cause
+    }
+  }
+}
+
+class AIGuard extends NoopAIGuard {
+  constructor (tracer, config) {
```

Comment:
we can also use `#` for all private properties/variables defined here (unless they're accessed in tests, in which case maybe getters make sense?)